snapshots/devmapper/pool_device.go

```diff
@@ -475,7 +477,8 @@ func (p *PoolDevice) RemoveDevice(ctx context.Context, deviceName string) error
 		return err
 	}
 
-	if err := p.deleteDevice(ctx, info); err != nil {
+	// If the device has been deleted already, it is fine.
+	if err := p.deleteDevice(ctx, info); err != nil && !errors.Is(err, unix.ENODATA) {
```

Comment:
Should we just return nil from `deleteDevice` if the device has been already deleted?

Reply:
I thought that the original strict behavior might be helpful for catchching mistakes (e.g. calling deleteDevice() against same device unintentionally). What do you think?

Reply:
Ping @mxpv

Reply:
Sorry for delay. `deleteDevice` is also called from `rollbackActivate`, I guess we don't want to fail rollback and add this check as well?

Reply:
Ah right. And having the same check on the two places don't look right... Okay. Let me put the check in deleteDevice() as you suggested.